libafl/src/corpus/testcase.rs

```diff
@@ -201,6 +212,280 @@ where
             ..Testcase::default()
         }
     }
+
+    /// Compute the `weight` used in weighted corpus entry selection algo
+    #[allow(clippy::cast_precision_loss, clippy::cast_lossless)]
+    pub fn compute_weight<S>(&self, state: &S) -> Result<f64, Error>
```

Comment:
I'm not sure that having this as a method of Testcase is correct. Ideally, we want it to be abstract allowing the user to define new weight calculations. Maybe it can be a generic parameter of the scheduler, like FavFctor is for the index minimizer scheduler.

Reply:
Btw we can merge as it is now (after fixing fuzzers/) so we can do experiments and then change it later

Reply:
so can I make it into a FavFactor<I, S>?

Reply:
y

Reply:
I'll do it in another PR then.